app/routes/task_routes.py

```diff
@@ -1 +1,132 @@
-from flask import Blueprint
\ No newline at end of file
+import os
+from flask import Blueprint, abort, make_response, request, Response
+from app.models.task import Task
+from datetime import datetime, timezone
+from ..db import db
+import requests
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+def validate_task(task_id):
+    
+    try:
+        tid = int(task_id)
+    except ValueError:
+        abort(make_response(
+            {"message": f"Task id '{task_id}' is invalid. Must be an integer."},
+            400
+        ))
+
+    task = db.session.get(Task, tid)
+    if not task:
+        not_found = {"message": f"Task with id '{task_id}' not found."}
+        abort(make_response(not_found, 404))
+
+    return task
+
+@tasks_bp.post("")
+def create_task():
+    request_body = request.get_json() or {}
+    if "title" not in request_body or "description" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)
+
+    new_task = Task.from_dict(request_body)
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_dict()}, 201
+
+@tasks_bp.get("")
+def get_all_tasks():
+
+    query = db.select(Task)
+
+    sort_param = request.args.get("sort") #Checks the URL for ?sort=asc or ?sort=desc
+    if sort_param == "asc":
+        query = query.order_by(Task.title)
+    elif sort_param == "desc":
+        query = query.order_by(Task.title.desc())
+    else:
+        query = query.order_by(Task.id)
+
+    tasks = db.session.scalars(query) #Executes the query
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append({
+            "id":          task.id,
+            "title":       task.title,
+            "description": task.description,
+            "is_complete": bool(task.completed_at)
+        })
```

Comment:
What about your `to_dict` method you wrote in `task.py`? Could we use that here?

```suggestion
    tasks_response = [task.to_dict for task in tasks]
```